sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py

```diff
@@ -23,7 +25,14 @@ def _signed_request(self, request):
 
         # Get the path and query from url, which looks like https://host/path/query
         query_url = str(request.http_request.url[len(self._credentials.host) + 8 :])
-
+        try:
+            if isinstance(request.context.transport, AioHttpTransport) or \
+                isinstance(getattr(request.context.transport, "_transport", None), AioHttpTransport) or \
+                isinstance(getattr(getattr(request.context.transport, "_transport", None), "_transport", None),
+                            AioHttpTransport):
```

Comment:
This looks weird. Could you add some comments?

Reply:
For context, this is pulled directly from [azure-storage-blob](https://github.com/Azure/azure-sdk-for-python/blob/d1ab444d9be412b9e93aaeb22b7df1f8a765e360/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py#L82). They have a similar issue with generating correct signatures from URLs when using `AioHttpTransport`, so they had implemented this (admittedly hacky) fix in the past.

Maybe we can refer to Storage's fix in a comment for this code, and explain the basics of the issue

Reply:
Yeah, that's referenced from storage, we need comments for the reference for sure. Will add. 

Some additional context, we need to use yarl.URL to get a correct encoding of key value in url from `%2A` to `*`, which caused some tests unauthorized while using `AioHttpTransport`.